server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -333,19 +338,48 @@ private MetaDataWithContent createMessageInOutbox(MessageWithId.CreationMessageE
     }
 
     private Flags getMessageFlags(CreationMessage message) {
+        Preconditions.checkState(!(hasKeywords(message) && isFlag(message)),
+                "Does not support both keywords and is*");
+        if (hasKeywords(message)) {
+            return buildFlagsFromKeywords(message.getKeywords().get());
+        } else {
+            return buildFlagsFromIsFlag(message);
+        }
+    }
+
+    private boolean isFlag(CreationMessage message) {
+        return message.isIsAnswered().isPresent()
+            || message.isIsDraft().isPresent()
+            || message.isIsFlagged().isPresent()
+            || message.isIsUnread().isPresent();
+    }
+
+    private boolean hasKeywords(CreationMessage message) {
+        return message.getKeywords().isPresent();
+    }
+
+    private Flags buildFlagsFromKeywords(ImmutableSet<JmapFlag> keywords) {
```

Comment:
because Flags is a very poorly designed object, we need to write a lot of convoluted code. Please get rid of Flags in jmap code as much as possible, there's no reason to leak the use of javax.mail in jmap.